chapter08/red/exercise3.pl

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+my %dirs = map {
+  opendir my $dir, $_ or die "Couldn't open $_: $!";
+  $_ => $dir
+} @ARGV;
```

Comment:
This is a nice elegant data structure, but I elected not to take the hash approach purely because I like to keep my hash keys alphanumeric.